yammbs/_minimize.py

```diff
@@ -120,7 +122,9 @@ class MinimizationResult(ImmutableModel):
 
 def _run_openmm(
     input: MinimizationInput,
-) -> MinimizationResult:
+) -> MinimizationResult | None:
```

Comment:
I'm neutral on polymorphism here, probably a bit negative on faking a failed system creation with a null-containing `MinimizationResult`, and mildly positive on a different object than `None` being returned, or there being a way to log it (possibly in provenance somewhere)

Reply:
Hmm, that seems interesting. I guess we could return `MinimizationResult | MinimizationError` where the error type holds some information that we'd want to put in the provenance, maybe the `qcarchive_id`? If it's just the id, maybe `| int` would be a lot simpler. Where were you picturing calling `_set_provenance`? With this approach it would be pretty nice to call it (or a related `_update_provenance`) in `optimize_mm`.

Also is there an easy way to match on the return type? Or do you just have to do `isinstance` or whatever? (I guess you can literally `match type(result)` but I don't know if that's actually idiomatic)

Reply:
I'm not sure what information you'd want in a `MinimizationError` class - I'd probably start with something like you're suggesting. (Feel free to just `return None` like you had, or whatever else you wish to try!)

Downstream I don't know if there's a better way to figure out if it's a successful or failed result; I can only think of `isinstance` checks or more fragile thinks like looking at particular attributes (either their existence or particular values).

Reply:
I think these changes will already fix the issue I was having (but I'd definitely be glad to add a test to make sure). Is the `MinimizationError` stuff something you'd want to see in this PR, or should that come later? I guess there's not as much rush if this is tied to the `yammbs-models` update anyway, but I was kind of hoping for a quick fix to unblock the CI test. I certainly don't want to rush through a half-baked idea that will cause problems later either, though.

Reply:
Up to you, but all things being equal I'd prefer whatever is minimal effort to get your issue fixed, so maybe that's just `return None`

Reply:
I realized I can just point my yammbs-dataset-submission env at this branch, so I'll see how it goes with this fix and we can think more about the ideal solution as yammbs-models develops.

Reply:
That sounds like a great finish line - I agree provenance would be nice but can be pushed into a future change